synapse/storage/databases/main/deviceinbox.py

```diff
@@ -639,10 +639,12 @@ def get_all_new_device_messages_txn(
             # Order by ascending stream ordering
             updates.sort()
 
-            limited = False
-            upto_token = current_id
+            upto_token = upper_pos
             if len(updates) >= limit:
                 upto_token = updates[-1][0]
```

Comment:
I'm not sure this `if` block makes sense any more.
We don't even limit the number of rows directly anyway.
I think `upper_pos` is always the right answer now?

Reply:
Hummm I am unsure if the right answer is `upper_pos` or `updates[-1][0]`.
I agree that the `if` seems useless however.

Using `upper_pos` should be fine as long as we are sure that the stream id actually exists already at this time, which should be always true I believe.

Reply:
I've use `upper_pos` (well, `upto_token` actually since I prefer this name, YMMV).
It simplifies the logic quite a bit.